From cf883172fd5c4f09d03b18fa54e4a6861d03c253 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:07:29 -0400
Subject: [tools] perf: Remove usage of trace_sched_wakeup(.success)

Message-id: <1410181752-25631-205-git-send-email-jolsa@redhat.com>
Patchwork-id: 91906
O-Subject: [PATCH RHEL7.1 BZ1134356 204/307] perf tools: Remove usage of trace_sched_wakeup(.success)
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 0680ee7db16de9c02d1d4b1a935a5daf754fe8a1
Author: Peter Zijlstra <peterz@infradead.org>
Date: Mon May 12 20:19:46 2014 +0200

description
===========
trace_sched_wakeup(.success) is a dead argument and has been for ages,
the only reason its still there is because of brain dead software, which
apparently includes perf tools

There's a few more instances in pearly snake shit, but that's not
supported as far as I care anyhow, so let that bitrot.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-sched.c b/tools/perf/builtin-sched.c
index 2579215..a3320f1 100644
--- a/tools/perf/builtin-sched.c
+++ b/tools/perf/builtin-sched.c
@@ -1007,17 +1007,12 @@ static int latency_wakeup_event(struct perf_sched *sched,
     struct perf_sample *sample,
     struct machine *machine)
 {
- const u32 pid   = perf_evsel__intval(evsel, sample, "pid"),
-    success = perf_evsel__intval(evsel, sample, "success");
+ const u32 pid   = perf_evsel__intval(evsel, sample, "pid");
  struct work_atoms *atoms;
  struct work_atom *atom;
  struct thread *wakee;
  u64 timestamp = sample->time;
 
- /* Note for later, it may be interesting to observe the failing cases */
- if (!success)
-  return 0;
-
  wakee = machine__findnew_thread(machine, 0, pid);
  atoms = thread_atoms_search(&sched->atom_root, wakee, &sched->cmp_pid);
  if (!atoms) {
diff --git a/tools/perf/tests/evsel-tp-sched.c b/tools/perf/tests/evsel-tp-sched.c
index 4774f7f..35d7fdb 100644
--- a/tools/perf/tests/evsel-tp-sched.c
+++ b/tools/perf/tests/evsel-tp-sched.c
@@ -74,9 +74,6 @@ int test__perf_evsel__tp_sched_test(void)
  if (perf_evsel__test_field(evsel, "prio", 4, true))
   ret = -1;
 
- if (perf_evsel__test_field(evsel, "success", 4, true))
-  ret = -1;
-
  if (perf_evsel__test_field(evsel, "target_cpu", 4, true))
   ret = -1;
 
-- 
1.7.1